tests/phpunit/api/v3/ContributionPageTest.php

```diff
@@ -1416,4 +1416,86 @@ public function addPriceFields(&$params) {
     );
   }
 
+  /**
+   * Test form submission CRM-16228.
+   */
+  public function testSubmitContributionPageWithPriceSetHavingTaxCRM16228() {
```

Comment:
Similar to above comments, it would be very helpful to rename this test to  "testContributionPageWithPriceSetHavingTaxHasCorrectTaxAndTotalAmounts" .